### PR TITLE
Update README file for gateway package

### PR DESCRIPTION
**Pull request by ZachEddy**
# Gateway README Refactor

This pull request updates the README file for the `gateway` package. I added a bit of extra documentation to the README, but I mostly just changed things around a little bit.

### CODE REVIEW

File: gateway/README.md [hunk 1]
```diff
@@ -105,9 +104,10 @@ func (s *myServiceImpl) MyMethod(ctx context.Context, req *MyRequest) (*MyRespon
 }
 ```
 
-If you do not use any custom outgoing header matcher you would see something like that:
+If you do not use any custom outgoing header matcher, you will see something like this.
+
 ```sh
-> curl -i http://localhost:8080/contacts/v1/contacts
+$ curl -i http://localhost:8080/contacts/v1/contacts
```
Comment (Evgeniy-L) [inline gateway/README.md]: Just a random thought. In contacts app we currently have `http://localhost:8080/v1/contacts`, but here it doesn't even matter for this example. I was thinking if it makes sense to change it to something neutral like `http://localhost:8080/data`. Up to you...

File: gateway/README.md [hunk 2]
```diff
@@ -118,55 +118,46 @@ Content-Length: 2
...
+	"github.com/infobloxopen/atlas-app-toolkit/gateway"
 )
 
 func init() {
 	forward_App_ListObjects_0 = gateway.ForwardResponseMessage
 }
 ```
 
-You can also refer [example app](https://github.com/github.com/infobloxopen/atlas-contacts-app/pb/contacts/contacts.overwrite.pb.gw.go)
+You can also refer [example app](https://github.com/infobloxopen/atlas-contacts-app/blob/master/pkg/pb/contacts.overwrite.pb.gw.go).
```
Comment (Evgeniy-L) [inline gateway/README.md]: This app is outdated but we can still refer it at least for this example. My concern is that some developers may consider using this app as an example. Do you think we should still refer old contacts app?

File: gateway/README.md [hunk 3]
```diff
@@ -118,55 +118,46 @@ Content-Length: 2
...
 
-#### Which forwarders I need to use to comply our REST API?
+#### Conforming to REST API Syntax
```
Comment (Evgeniy-L) [inline gateway/README.md]: `REST API Syntax` phrase may be confusing for someone from outside Infoblox. Not sure if we should be concerned about it :)

File: gateway/README.md (hunk 2 above)
Comment (ZachEddy) [reply gateway/README.md]: Agreed. I'll remove this sentence.

**Pull request merged**